src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -64,6 +64,22 @@ bool ReplicatedMergeTreeQueue::isVirtualPart(const MergeTreeData::DataPartPtr &
     return !virtual_part_name.empty() && virtual_part_name != data_part->name;
 }
 
+bool ReplicatedMergeTreeQueue::checkPartInQueueAndGetSourceParts(const String & part_name, Strings & source_parts) const
+{
+    std::lock_guard lock(state_mutex);
+
+    for (const auto & entry : queue)
+    {
+        if (entry->new_part_name == part_name)
+        {
+            source_parts.insert(source_parts.end(), entry->source_parts.begin(), entry->source_parts.end());
+            return true;
```

Comment:
It will not work correctly in case when current replica was recently cloned. Queue may contain multiple entries with the same `new_part_name`: one original entry (of any type) and multiple `GET_PART` entries.
https://github.com/ClickHouse/ClickHouse/blob/bfc122df64a96abbcefc7203dfe9068e98050513/src/Storages/StorageReplicatedMergeTree.cpp#L2828-L2839

Reply:
But it should be Ok in our case? Source parts of the old entry will be checked and the new `GET_PART` entry will be replaced with the empty part it is lost. Maybe I have to update the comment.